index.html

```diff
@@ -191,48 +281,116 @@ <h2>Modifications to existing procedures</h2>
+            </li>
+            <li>
+              <p>Return
+              <var>transceiver</var>.{{RTCRtpTransceiver/[[HeaderExtensionsToNegotiate]]}}.</p>
+            </li>
+          </ol>
+        </dd>
+      </dl>
+      <dl>
+        <dt><dfn data-dfn-for=RTCRtpTransceiver>setHeaderExtensionsToNegotiate</dfn></dt>
         <dd>
-          Execute the following steps:
+          <p>Execute the following steps:</p>
           <ol>
-            <li>Let <var>extlist</var> be the list of extensions in the argument.</li>
             <li>
-              For each extension <var>ext</var> in <var>extlist</var>, do the following:
+              <p>Let <var>transceiver</var> be the {{RTCRtpTransceiver}} that
+              this method was invoked on.</p>
+            </li>
+            <li>
+              <p>Let <var>extensions</var> be the first argument of this
+              method.</p>
+            </li>
+            <li>
+              <p>If the size of <var>extensions</var> does not match the size of
+              <var>transceiver</var>.{{RTCRtpTransceiver/[[HeaderExtensionsToNegotiate]]}}
+              [=exception/throw=] an {{InvalidModificationError}} and return
+              <code>undefined</code>.
+              </p>
+            </li>
+            <li>
+              <p>For each index <var>i</var> of <var>extensions</var>, run the
+              following steps:</p>
               <ol>
                 <li>
-                  If <var>ext</var>.{{RTCRtpHeaderExtensionCapability/uri}} value is missing, [=exception/throw=] a {{TypeError}}.
+                  <p>Let <var>extension</var> be the <var>i</var>-th element of
+                  <var>extensions</var>.</p>
                 </li>
-                <li>If <var>ext</var>.{{RTCRtpHeaderExtensionCapability/uri}} value is not present in
-                  {{RTCRtpTransceiver/[[HeaderExtensionsToOffer]]}}, [=exception/throw=] a {{NotSupportedError}}.
+                <li>
+                  <p>If
+                  <var>extension</var>.{{RTCRtpHeaderExtensionCapability/uri}}
+                  or
+                  <var>extension</var>.{{RTCRtpHeaderExtensionCapability/direction}}
+                  is missing, [=exception/throw=] a {{TypeError}} and return
+                  <code>undefined</code>.</p>
                 </li>
                 <li>
-                  Let <var>entry</var> be the entry with the same {{RTCRtpHeaderExtensionCapability/uri}} value in
-                  {{RTCRtpTransceiver/[[HeaderExtensionsToOffer]]}}.
+                  <p>If
+                  <var>extension</var>.{{RTCRtpHeaderExtensionCapability/uri}}
+                  is not equal to the {{RTCRtpHeaderExtensionCapability/uri}} of
+                  the <var>i</var>-th element of
+                  <var>transceiver</var>.{{RTCRtpTransceiver/[[HeaderExtensionsToNegotiate]]}},
+                  [=exception/throw=] an {{InvalidModificationError}} and return
+                  <code>undefined</code>.</p>
```

Comment:
Why returning undefined after throwing.
We should just abort these steps.

Reply:
I'll update to throw